4.8.0 licensing issues.

joshelser opened the pull request:
Fixes for the Apache Phoenix source release

JamesRTaylor commented on examples/STOCK_SYMBOL.sql:
```diff
@@ -1,3 +1,19 @@
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
```
There was a sqlline bug which prevented SQL scripts from running where the first line was a comment. Would you mind confirming that this has been fixed in the version of sqlline we're using? If it's a problem, we can just remove these examples as they're not particularly important.

mujtabachohan replied:
This is not fixed in sqlline 1.1.9 that we are using. Fix will be part of yet unreleased 1.1.10. However this works fine using psql.py script.

joshelser replied:
Great. Thanks for the catch! I totally did not use it. Let me re-add these to the exclusions and make an issue to track this information.

JamesRTaylor commented on LICENSE:
```diff
@@ -200,3 +200,90 @@
+
+---
+
+This product bundles portions of AngularJS (https://angularjs.org/) which
```
The only reason we bundle this project and the ones below it are for the trace UI. If this is going to make it more difficult to do a binary release or require more work for the source releases, I think we should just remove the trace UI from both source and binary releases.

joshelser replied:
The text added here should be all that is needed for 4.8.0's source release for the trace ui stuff. The accompanying binary artifact just needs the same text copied into its LICENSE and NOTICE. While we can move this out for 4.8.0, I think I'd say we can decide what to do in a 4.9.0 (does this code get dropped, moved into some other repo, etc). We certainly can just remove it now if desired.

JamesRTaylor replied:
What about the missing ASF header for the javascript files?

joshelser replied:
Oh right. That was done in 7f54cd8. Still though, that was straightforward to fix.

JamesRTaylor commented:
Thanks, @joshelser! Couple of minor comments. Really appreciate your efforts here.

joshelser commented:
7d9adab removes the license headers from `examples/**/*.sql` and re-adds the exclusions to apache-rat-plugin. Filed https://issues.apache.org/jira/browse/PHOENIX-3099 to track it.

joshelser commented:
Ok, I think this one is good to go. Any chance I could bug some folks to glance over these changes (I know they're massive). @billierinaldi, @joewitt, @apurtell, @busbey

FYI @JamesRTaylor and @ankitsinghal (to know things are close).

busbey commented on NOTICE:
```diff
@@ -1,37 +1,18 @@
 
 The file bin/daemon.py is based on the file of the same name in python-daemon 2.0.5
-(https://pypi.python.org/pypi/python-daemon/) distributed under the terms of
-the Apache 2 license. Original source copyright:
```
what's the license for this inclusion? I presume not Apache 2, since it's removed?

joshelser replied:
No, it is ASLv2. Omitting because it's unnecessary to state because it is ASLv2. Just propagating the copyright information.

busbey replied:
✅

busbey commented on NOTICE:
```diff
@@ -1,37 +1,18 @@
+This also includes:
 
 The phoenix-spark module has been adapted from the phoenix-spark library
 distributed under the terms of the Apache 2 license. Original source copyright:
```
nit: "Apache License version 2.0"

joshelser replied:
I suppose under the same logic as the daemon.py entry, I could drop that too. AFAIK, these were not verbatim "copies" from other NOTICE files, just stating that they were copied and not original works.

busbey replied:
So long as it's either removed or named consistently, ✅

busbey commented on dev/release_files/LICENSE:
```diff
@@ -201,28 +201,1238 @@
+---
+
+This product contains the following products which are licensed under the Common
+Development and Distribution License (CDDL).
```
which version of the CDDL?

joshelser replied:
Good catch, let me dbl check and fix.

joshelser replied:
Oh, shame. I also didn't include the text for CDDL

busbey replied:
good catch on your part then, I didn't notice that bit. :)

joshelser replied:
Fixed in cac336a (will push shortly)

busbey commented on dev/release_files/LICENSE:
```diff
@@ -201,28 +201,1238 @@
+
+---
+
+This product includes the following products which are in the public domain
```
nit: is there a difference between "public domain" here and "Public Domain" back up above?

joshelser replied:
Nope omission on my part. 4ade807

busbey commented on dev/release_files/LICENSE:
```diff
@@ -201,28 +201,1238 @@
+        REGARDING THE INFORMATION PROVIDED, AND DISCLAIMS LIABILITY FOR
+        DAMAGES RESULTING FROM ITS USE.
+
+  See below for full CCPL License.
```
should be the CC-A license, I think?

joshelser replied:
I believe this was a verbatim copy from the shipped hbase NOTICE files. Maybe we need correction there?

busbey replied:
hurm. which branch?

[current master LICENSE for that bit just has the full text](https://github.com/apache/hbase/blob/6840c5fa6cdeb2409d21fbaed173f694968fdd0a/LICENSE.txt#L293) and [current master NOTICE looks very different](https://github.com/apache/hbase/blob/6840c5fa6cdeb2409d21fbaed173f694968fdd0a/NOTICE.txt#L14) (and probably needs to be cleaned up).

joshelser replied:
Fixed in 3120212

busbey commented on dev/release_files/LICENSE:
```diff
@@ -201,28 +201,1238 @@
+  This project contains annotations in the package org.apache.http.annotation
+  which are derived from JCIP-ANNOTATIONS
+  Copyright (c) 2005 Brian Goetz and Tim Peierls.
+  See http://www.jcip.net and the Creative Commons Attribution License
```
Above we say "Creative Commons Public License" and here "Creative Commons Attribution", which is it?

joshelser replied:
CC by A 2.5 is correct. The little header was me being dumb.

busbey commented on dev/release_files/LICENSE:
```diff
@@ -201,28 +201,1238 @@
+    http://jquery.com/
+
+  This project bundles a derivative image for our Orca Logo. This image is
+  available under the Creative Commons By Attribution 3.0 License.
```
I don't see the text of the Create Commons By Attribution 3.0 license in the diff. Was it already in the base file or something?

joshelser replied:
No, it wasn't. I think I missed the full license text here too. hbase-server.jar might also need looking at for the same reason (mental note).

busbey replied:
ugh. yeah let me know if I have to dig. That jar gets filled in via [a velocity template macro](https://github.com/apache/hbase/blob/ce657d591114d1dab4aa4cf65698dfbae29406e3/hbase-resource-bundle/src/main/resources/META-INF/LICENSE.vm#L264), so if that's not working correctly I'll have some problems.

(also, jeez I forgot how involved the HBase solution to this is)

joshelser replied:
Haha, ok, I think hbase-server.jar. It was a bad copy-paste job from me.

joshelser commented:
Thanks again for taking a look at this, Sean! You found some really nice brain farts from me.

Changes made:
* Clarified CDDL 1.0 and 1.1 references in LICENSE
* Included CDDL 1.0 and 1.1 text in LICENSE
* Corrected the LICENSE mentions for hbase-server (bad copy-paste)
* Clarified CC by A 2.5 and 3.0 products in LICENSE
* Included CC by A 2.5 and 3.0 texts in LICENSE
* Remove an unnecessary mention to ASLv2 in NOTICE
* Removed all trailing whitespace in L&N

joshelser commented:
I think this is good to go. Does anyone else want to look at this before I merge it in?

apurtell commented on dev/make_rc.sh:
```diff
@@ -30,10 +30,10 @@ DIR_ROOT="$(cd $(dirname $0);pwd)/.."
 DIR_REL_BASE=$DIR_ROOT/release
-DIR_REL_ROOT=$DIR_REL_BASE/phoenix-$PHOENIX
-DIR_REL_BIN=phoenix-$PHOENIX-bin
+DIR_REL_ROOT=$DIR_REL_BASE/apache-phoenix-$PHOENIX
```
This will be fun for repackagers, but it's something that can be handled with a regex

apurtell commented on dev/make_rc.sh:
```diff
@@ -88,15 +88,15 @@ cp $DIR_DOCS/* $DIR_REL_BIN_PATH;
 
 # Generate bin tar
-tar cvzf $DIR_REL_BIN_TAR_PATH/$DIR_REL_BIN.tar.gz -C $DIR_REL_ROOT phoenix-$PHOENIX-bin;
+tar cvzf $DIR_REL_BIN_TAR_PATH/$DIR_REL_BIN.tar.gz -C $DIR_REL_ROOT apache-phoenix-$PHOENIX-bin;
```
This will be fun for repackagers but can be handled by a regex I suppose

apurtell commented:
lgtm

joshelser commented:
@apurtell I got some emails from you on the naming of the tarballs, but I'm struggling to find the thread here on GH. I'm not actually 100% positive if the artifacts being named "phoenix" and not "apache-phoenix" is against policy. I would be OK reverting that change if it's not actually required that the names are "apache-foo" (and only recommended) -- seeing if I can find the relevant policy docs.

joshelser commented:
Thanks for all of the help everyone. This was merged.
